### PR TITLE
js: functional and modern syntax

### PR DESCRIPTION
**Pull request by tsao-chi**
generate more elegant code

it get rid of
```
function(...){
return function(...){
return function(...){
return function(...){
...
```

### CODE REVIEW

File: src/transpilers/js.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export default class JSCompiler extends BaseTranspiler {
             } else if (a.type == "bol") {
               value = "false";
             } else if (a.type == "fun") {
-              value = "()=>0";
+              value = "null";
```
Comment (tsao-chi) [inline src/transpilers/js.ts]: (users shoudn't call a function before the function is assigned)

**Comment by LingDong-** [comment]
Looking great, let me know if it's ready for merging, thanks!

**Comment by tsao-chi** [comment]
it's ready for merging

**Comment by tsao-chi** [comment]
it's ready for merging

**Comment by LingDong-** [comment]
Thanks!

I see you did a `x=>y=>z=>{}` to get rid of `funcurlvls`, pretty neat.

I'm thinking if functions should still be initialized to `()=>0`. All other types initialize to a value. Numbers to `0`, Bools to `false`, Objects to `{}`. I know user shouldn't need to use a function before it is defined, but let's not make assumptions. This also makes runtime typechecking more consistent.

One thing that is debatable is that maybe we can return `_=>_` the identity function instead. I guess either is quite reasonable.

What do you think?

**Comment by tsao-chi** [comment]
I think `_=>{}` is more reasonable

**Comment by LingDong-** [comment]
Thanks a lot!

**Comment by tsao-chi** [comment]
@LingDong- Screenshots and CHANGELOG need to be updated

**Pull request merged**